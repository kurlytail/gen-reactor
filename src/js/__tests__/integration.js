import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/pom.min.js -h').toString();
        expect(output).toMatchSnapshot();
    });
    it('## should generate design and run pom commands with reactor extensions', () => {
        let output = execSync(
            'sgen -g pom -g `pwd`/dist/reactor.min.js -d src/test/fixture/design.json -o testoutput'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync('npm install', { cwd: 'testoutput' }).toString();
        output = execSync('npm run lint', { cwd: 'testoutput' }).toString();
        output = execSync('mvn compile', { cwd: 'testoutput' }).toString();
    });
});
