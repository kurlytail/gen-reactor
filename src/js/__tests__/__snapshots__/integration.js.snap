// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design and run reactor commands 1`] = `
"
info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
  -g, --generator=ARG+ generators to be used
  -h, --help           display this help
"
`;
